storage/mempool.go

```diff
@@ -185,6 +185,45 @@ func (s *Storage) GetPendingUserState(stateID uint32) (*models.UserState, error)
 	}, nil
 }
 
+// does not return the full list of states, only the pending c2ts
+func (s *Storage) GetPendingUserStates(pubkey *models.PublicKey) ([]models.UserState, error) {
```

Comment:
Little bit confusing naming with `GetPendingUserState`, but let's not bikeshed on that now.